### PR TITLE
Gha deploy release

### PR DESCRIPTION
**Pull request by esc**
replaces #52

### CODE REVIEW

**Comment by esc** [comment]
I reckon the secrets are now *not* being scrubbed as this is PR from the 'blessed' repo (non-fork). Still getting the following from PyPi servers though:

`Received "502: Bad Gateway" Package upload appears to have failed.  Retry 1 of 5``

It retries then 5 times but doesn't succeed.

**Comment by esc** [comment]
Using tine locally, I can reproduce this, sometimes I get:

```
Uploading distributions to https://upload.pypi.org/legacy/
Uploading numba-scipy-0.3.0.dev0+20.gba039a54f3.tar.gz
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 28.1k/28.1k [00:01<00:00, 14.7kB/s]
NOTE: Try --verbose to see response content.
HTTPError: 400 Bad Request from https://upload.pypi.org/legacy/
'0.3.0.dev0+20.gba039a54f3' is an invalid value for Version. Error: Can't use PEP 440 local versions. See https://packaging.python.org/specifications/core-metadata for more information.
```

and sometimes I get:

```
Uploading distributions to https://upload.pypi.org/legacy/
Uploading numba-scipy-0.3.0.dev0+20.gba039a54f3.tar.gz
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 28.1k/28.1k [00:02<00:00, 12.0kB/s]
Received "502: Bad Gateway" Package upload appears to have failed.  Retry 1 of 5
Uploading numba-scipy-0.3.0.dev0+20.gba039a54f3.tar.gz
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 28.1k/28.1k [00:01<00:00, 20.0kB/s]
Received "502: Bad Gateway" Package upload appears to have failed.  Retry 2 of 5
Uploading numba-scipy-0.3.0.dev0+20.gba039a54f3.tar.gz
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 28.1k/28.1k [00:01<00:00, 21.8kB/s]
Received "502: Bad Gateway" Package upload appears to have failed.  Retry 3 of 5
Uploading numba-scipy-0.3.0.dev0+20.gba039a54f3.tar.gz
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 28.1k/28.1k [00:01<00:00, 23.2kB/s]
Received "502: Bad Gateway" Package upload appears to have failed.  Retry 4 of 5
Uploading numba-scipy-0.3.0.dev0+20.gba039a54f3.tar.gz
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 28.1k/28.1k [00:01<00:00, 22.1kB/s]
Received "502: Bad Gateway" Package upload appears to have failed.  Retry 5 of 5
NOTE: Try --verbose to see response content.
HTTPError: 502 Bad Gateway from https://upload.pypi.org/legacy/
Bad Gateway
```

for the same command! My guess that this varies depending on which backend server I end up at.

**Comment by esc** [comment]
So, I guess we can merge a version of this that is  triggered by tags only, and assume that on the next tag it will be fine and upload to PyPi.

**Comment by esc** [comment]
@stuartarchibald this is now "ready for review".

File: .github/workflows/deploy_release.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  deploy_pypi:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Setup Miniconda
+        uses: conda-incubator/setup-miniconda@v2.0.1
```
Comment (stuartarchibald) [inline .github/workflows/deploy_release.yml:18]: Is the `.0.1` important? the `test_and_deploy.yml` uses `v2`.

Comment (esc) [reply .github/workflows/deploy_release.yml:18]: no, this is just the latest version, I tried it during debugging.I don't think it matters much. In the end we will most likely be using https://dependabot.com/github-actions/ to keep this kind of stuff up-to-date.

**Review by stuartarchibald — COMMENTED** [review]
Thanks for the patch, one minor question else looks good.

File: .github/workflows/deploy_release.yml (hunk 1 above)
Comment (stuartarchibald) [reply .github/workflows/deploy_release.yml:18]: ok, thanks for confirming.

**Review by stuartarchibald — APPROVED** [review]
Thanks for the patch.

**Pull request merged**